sdk/daml-lf/parser/src/test/scala/com/digitalasset/daml/lf/testing/parser/ParsersSpec.scala

```diff
@@ -563,7 +564,7 @@ class ParsersSpec(majorLanguageVersion: LanguageMajorVersion)
         """
           module Mod {
 
-            variant Tree (a : * ) = Leaf : Unit | Node : Mod:Tree.Node a ;
```

Comment:
Is this change supposed to be part of this PR?

Reply:
This acts as a regression test, without the parser change, this is parsed improperly. 

concretely when using simple `|` for type synonyme  (i.e. without the parser change from this PR), 
the new test is read improperly,  i.e.
`Node : Mod:Tree.Node a | Leaf : Unit | Undefined : Unit` is read as 
`Node : (Mod:Tree.Node) a (| Leaf : Unit |) (Undefined:Unit)`
a single constructor `Node` with type  `(Mod:Tree.Node) a (|Leaf:Unit|) (Undefined:Unit)`


Note that 

`Leaf : Unit | Node : Mod:Tree.Node a |  Undefined : Unit` is read correctly because 
`Node : Mod:Tree.Node a` cannot be read as a type.